include/taskolib/execute_lua_script.h

```diff
@@ -32,6 +32,16 @@
 
 namespace task {
 
+/**
+ * Check the syntax of a Lua script without running it.
+ *
+ * \returns an empty string if the script is syntactically correct, or an error message
+ *          otherwise. The error message is pre-processed to a certain degree: Unhelpful
+ *          parts like the chunk name of the script (`[string "..."]:`) are removed, and a
+ *          few known special messages are replaced by more readable explanations.
+ */
+std::string check_lua_script(sol::state& lua, sol::string_view script);
```

Comment:
I would propose to throw an `Error` exception instead of return the result string. if it detects a syntax error it is caught in the caller: `void check_lua_script(sol::state&, sol::string_view)`. Seems more natural to me.

Reply:
I understand that, but on the other hand it would force users to write local try-catch blocks at least in some cases:
```cpp
try 
{
    check_lua_script(lua, script);
}
catch (const std::exception& e) 
{
    report_error(e.what());
}
```
That is quite complicated and wasteful if all you really want is
```cpp
auto err_msg = check_lua_script(lua, script);
if (not err_msg.empty())
    report_error(err_msg);
```

Reply:
We could return `gul14::expected<sol::load_result, std::string>` and call the function `load_lua_script()` instead:
```cpp
auto load_result = load_lua_script(lua, script);
if (!load_result)
    report_error(load_result.error());
```
That would have the additional benefit of letting users do something with the loaded script if they really want to (like execute it later on).

Reply:
Yep ... using `gul14::expected<..>` will be the optimal solution: small but expressive :-)

Reply:
Implemented now.